app/controllers/search.controller.js

```diff
@@ -1,55 +1,136 @@
-    numberOfLinks: 0,
-    totalAdWordsAdvertisers: 0,
-    rawHTML: 'Error',
-  };
-
+  // Crawl data from each keyword
   for (const keyword of keywords) {
-    await page.goto(`http://www.google.com/search?q=${keyword}&hl=en`, {
+    // initial object
+    const data = {
+      keyword: keyword.keyword,
+      userId: keyword.userId,
+      totalResultsOfKeyword: '0',
+      numberOfLinks: 0,
+      totalAdWordsAdvertisers: 0,
+      rawHTML: '',
+      isCompleted: false,
+    };
+    // wait 20s to avoid Google captcha if possible
+    await timer(20000);
+
+    await page.goto(`http://www.google.com/search?q=${data.keyword}&hl=en`, {
       waitUntil: 'networkidle2',
     });
-    data.keyword = keyword;
-    data.totalResultsOfKeyword = await page.evaluate(
-      'document.querySelector("#result-stats").innerText'
-    );
-    data.numberOfLinks = await page.$$eval(
-      'a',
-      (as) => as.map((a) => a.href).length
-    );
-    data.totalAdWordsAdvertisers = await page.evaluate(
-      () => document.querySelectorAll('[aria-label="Ads"] > div').length
-    );
-    data.rawHTML = await page.evaluate(
-      () => document.getElementsByTagName('html')[0].innerHTML
-    );
+    try {
+      data.totalResultsOfKeyword = await page.evaluate(
+        'document.querySelector("#result-stats").innerText'
+      );
+      data.numberOfLinks = await page.$$eval(
+        'a',
+        (as) => as.map((a) => a.href).length
+      );
+      data.totalAdWordsAdvertisers = await page.evaluate(
+        () => document.querySelectorAll('[aria-label="Ads"] > div').length
+      );
+      data.rawHTML = await page.evaluate(
+        () => document.getElementsByTagName('html')[0].innerHTML
+      );
+      data.isCompleted = true;
+      Keyword.update(
+        {
+          ...data,
+        },
+        {
+          where: {
+            id: keyword.id,
+          },
+        }
+      );
+    } catch (e) {}
+  }
+
+  browser.close();
+
+  // Run cron too check if the account have in-complete keyword.
+  const task = cron.scheduleJob('*/1 * * * * *', async () => {
+    await crawlKeyword(userId);
+    task.cancel();
+  });
+});
 
-    await Keyword.create({
-      ...data,
+async function crawlKeyword(userId) {
+  Keyword.findAll({
+    where: {
+      userId: userId,
+    },
+    order: [['id', 'DESC']],
+  }).then(async (result) => {
+    if (Boolean(result.length)) {
+      let keyWordUnCrawl = [];
+      for (const keyword of result) {
+        if (!keyword.isCompleted) {
+          keyWordUnCrawl.push(keyword);
+        }
+      }
+      keywordManager.crawl(userId, keyWordUnCrawl);
+    }
+  });
+}
+
+exports.searchByEachKeyWords = async (req, res) => {
+  const { keywords } = req.body;
+
+  if (keywords.length <= 100 && Boolean(keywords.length)) {
+    let arrayValidated = keywords
+      .filter((keyword) => keyword)
+      .map((keyword) => ({
+        keyword: keyword,
+        userId: req.userId,
+        totalResultsOfKeyword: 'Error',
```

Comment:
Not sure why setting it to a string that says `Error`

Reply:
It fixed!